graylog2-server/src/main/java/org/graylog2/alerts/AlertImpl.java

```diff
@@ -16,89 +16,91 @@
  */
 package org.graylog2.alerts;
 
-import com.google.common.collect.Maps;
-import org.bson.types.ObjectId;
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
 import org.graylog2.database.CollectionName;
-import org.graylog2.database.PersistedImpl;
-import org.graylog2.plugin.Tools;
-import org.graylog2.plugin.database.validators.Validator;
 import org.joda.time.DateTime;
-import org.joda.time.DateTimeZone;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.mongojack.Id;
+import org.mongojack.ObjectId;
 
-import java.util.Collections;
+import java.util.Date;
 import java.util.Map;
 
+@AutoValue
+@JsonAutoDetect
 @CollectionName("alerts")
-public class AlertImpl extends PersistedImpl implements Alert {
-    private static final String FIELD_ID = "_id";
-    private static final String FIELD_CONDITION_ID = "condition_id";
-    private static final String FIELD_STREAM_ID = "stream_id";
-    private static final String FIELD_DESCRIPTION = "description";
-    private static final String FIELD_CONDITION_PARAMETERS = "condition_parameters";
-    private static final String FIELD_TRIGGERED_AT = "triggered_at";
-
-    private static final Logger LOG = LoggerFactory.getLogger(AlertImpl.class);
-
-    public static final int MAX_LIST_COUNT = 300;
-    public static final int REST_CHECK_CACHE_SECONDS = 30;
-
-    protected AlertImpl(Map<String, Object> fields) {
-        super(fields);
-    }
-
-    protected AlertImpl(ObjectId id, Map<String, Object> fields) {
-        super(id, fields);
-    }
-
-    public Map<String,Object> toMap() {
-        Map<String, Object> map = Maps.newHashMap();
-
-        DateTime triggeredAt = new DateTime(fields.get(FIELD_TRIGGERED_AT), DateTimeZone.UTC);
-
-        map.put("id", fields.get(FIELD_ID).toString());
-        map.put(FIELD_CONDITION_ID, fields.get(FIELD_CONDITION_ID));
-        map.put(FIELD_STREAM_ID, fields.get(FIELD_STREAM_ID));
-        map.put(FIELD_DESCRIPTION, fields.get(FIELD_DESCRIPTION));
-        map.put(FIELD_CONDITION_PARAMETERS, fields.get(FIELD_CONDITION_PARAMETERS));
-        map.put(FIELD_TRIGGERED_AT, Tools.getISO8601String(triggeredAt));
-
-        return map;
-    }
+public abstract class AlertImpl implements Alert {
```

Comment:
We do that with most value/dao classes to keep the implementation separate and program to interfaces, and I think it makes sense to keep it. Especially for this PR to be consistent with the rest.